Library/Homebrew/utils/ruby.sh

```diff
@@ -0,0 +1,46 @@
+setup-ruby-path() {
+  local HOMEBREW_VENDOR_RUBY_OPT;
+  local HOMEBREW_VENDOR_RUBY_PATH;
+
+  if [[ -z "$HOMEBREW_DEVELOPER" ]]
+  then
+    unset HOMEBREW_RUBY_PATH
+  fi
+
+  if [[ -z "$HOMEBREW_RUBY_PATH" && "$HOMEBREW_COMMAND" != "install-vendor" ]]
+  then
+    HOMEBREW_VENDOR_RUBY_OPT="$HOMEBREW_LIBRARY/Homebrew/vendor/ruby/opt"
+    HOMEBREW_VENDOR_RUBY_PATH="$HOMEBREW_VENDOR_RUBY_OPT/bin/ruby"
+    if [[ -x "$HOMEBREW_VENDOR_RUBY_PATH" ]]
+    then
+      HOMEBREW_RUBY_PATH="$HOMEBREW_VENDOR_RUBY_PATH"
+
+      if [[ $(readlink "$HOMEBREW_VENDOR_RUBY_OPT") != "$(<"$HOMEBREW_LIBRARY/Homebrew/vendor/ruby-version")" ]]
+      then
+        if ! brew install-vendor ruby --quiet
+        then
+          onoe "Failed to upgrade vendor Ruby."
+        fi
+      fi
+    else
+      if [[ -n "$HOMEBREW_OSX" ]]
+      then
+        HOMEBREW_RUBY_PATH="/System/Library/Frameworks/Ruby.framework/Versions/Current/usr/bin/ruby"
+      else
+        HOMEBREW_RUBY_PATH="$(which ruby)"
+      fi
+
+      if [[ -z "$HOMEBREW_RUBY_PATH" || "$("$HOMEBREW_RUBY_PATH" -e "puts RUBY_VERSION.split('.').first")" != "2" ]]
```

Comment:
Could we use `ruby --version` and parse the output with some Bash built-ins instead of making it execute actual Ruby code to get the major version? `ruby --version` is significantly faster on my machine than `ruby -e "<some-trivial-code>"`. (Still, finding a reliable way to completely omit this check on every run would be great.)